packages/optimizer/spec/transformers/valid/ServerSideRendering/generates_unique_ids_for_items_with_media_and_ignores_items_with_own_id/expected_output.html

```diff
@@ -6,13 +6,13 @@
 </head>
 <body>
   <amp-img height="100" width="100" layout="responsive" src="img1.png" class="i-amphtml-layout-responsive i-amphtml-layout-size-defined" i-amphtml-layout="responsive" id="i-amp-0">
-    <i-amphtml-sizer slot="i-amphtml-svc" style="display:block;padding-top:100%"></i-amphtml-sizer>
+    <i-amphtml-sizer slot="i-amphtml-svc" style="display:block"></i-amphtml-sizer>
   </amp-img>
   <amp-img id="customId" height="100" width="100" layout="responsive" src="img2.png" class="i-amphtml-layout-responsive i-amphtml-layout-size-defined" i-amphtml-layout="responsive">
-    <i-amphtml-sizer slot="i-amphtml-svc" style="display:block;padding-top:100%"></i-amphtml-sizer>
+    <i-amphtml-sizer slot="i-amphtml-svc" style="display:block"></i-amphtml-sizer>
   </amp-img>
   <amp-img height="100" width="100" layout="responsive" src="img3.png" class="i-amphtml-layout-responsive i-amphtml-layout-size-defined" i-amphtml-layout="responsive" id="i-amp-1">
-    <i-amphtml-sizer slot="i-amphtml-svc" style="display:block;padding-top:100%"></i-amphtml-sizer>
+    <i-amphtml-sizer slot="i-amphtml-svc" style="display:block"></i-amphtml-sizer>
```

Comment:
The elements in [`input.html`](https://github.com/ampproject/amp-toolbox/blob/337a494a293b672d5918dbe3bb8e6c868b150b03/packages/optimizer/spec/transformers/valid/ServerSideRendering/generates_unique_ids_for_items_with_media_and_ignores_items_with_own_id/input.html#L7-L9) do not have `heights` attributes. Is this change properly being limited to when `heights` is present?

Reply:
In [amphtml source](https://github.com/ampproject/amphtml/blob/main/src/core/static-layout.js#L155) I've found when the padding-top is being set. I've update the code and now padding-top will set when `heights` attribute is missing.

Reply:
I don't see there in `static-layout.js` where the presence of the `heights` attribute determines whether `padding-top` is added. Maybe when the `heights` attribute is processed by the runtime, the `padding-top` later is removed/overridden? This appears so:

https://github.com/ampproject/amphtml/blob/160dcd9c93d34bb7da073a1240c287fc0cf14591/src/custom-element.js#L1068-L1078

Reply:
Sorry for the confusion. I wanted to say the same. Yes, you're right. [applyStaticLayout](https://github.com/ampproject/amphtml/blob/160dcd9c93d34bb7da073a1240c287fc0cf14591/src/custom-element.js#L1205) creates the sizer element first, then adds the padding-top based on the height and width attributes. Later it [gets overridden](https://github.com/ampproject/amphtml/blob/160dcd9c93d34bb7da073a1240c287fc0cf14591/src/custom-element.js#L1206) if the heights attribute is present.